public/app/types/dashboard.ts

```diff
@@ -39,6 +39,7 @@ export interface DashboardMeta {
   hasUnsavedFolderChange?: boolean;
   annotationsPermissions?: AnnotationsPermissions;
   publicDashboardAccessToken?: string;
+  hasPublicDashboard?: boolean;
```

Comment:
nit: We need to think if a Dashboard HAS a Public Dashboard or if a Dashboard IS public or not. Maybe this varibale should be named as `isPublic` but I'm not sure.

Reply:
In the docs, we are saying things like `make a dashboard public` or `share your Grafana dashboard with anyone`. So it seems it is an attribute of a Grafana Dashboard and a flag `isPublic` could be more appropriate.

Reply:
maybe now that we have the access token around its less ambiguous? Idk

Reply:
That's true, because we use the same model for both. Maybe we need another flag that indicates how a dashboard is being rendered, in a `public` view mode or an `auth` view because at the end, it is the same dashboard. But you are right, in this case `hasPublicDashboard` makes sense to me too.

Reply:
I believe we have another variable somewhere for IsPublicView or similar. I would check for access token and public dashboard view